flang/test/Lower/OpenMP/omp-wsloop-collapse.f90

```diff
@@ -0,0 +1,240 @@
+! This test checks lowering of OpenMP DO Directive(Worksharing) with collapse.
+
+! RUN: bbc -fopenmp -emit-fir %s -o - | \
+! RUN:   FileCheck %s --check-prefix=FIRDialect
+! RUN: bbc -fopenmp -emit-fir %s -o - | \
+! RUN:   tco --disable-llvm --print-ir-after=fir-to-llvm-ir 2>&1 | \
+! RUN:   FileCheck %s --check-prefix=LLVMIRDialect
+! RUN: bbc -fopenmp -emit-fir %s -o - | \
+! RUN:   tco | FileCheck %s --check-prefix=LLVMIR
+
+program wsloop_collapse
+! LLVMIRDialect-LABEL:   llvm.func @_QQmain() {
+  integer :: i, j, k
+  integer :: a, b, c
+  integer :: x
+! FIRDialect:         %[[VAL_0:.*]] = fir.alloca i32 {bindc_name = "a", uniq_name = "_QEa"}
+! FIRDialect:         %[[VAL_1:.*]] = fir.alloca i32 {bindc_name = "b", uniq_name = "_QEb"}
+! FIRDialect:         %[[VAL_2:.*]] = fir.alloca i32 {bindc_name = "c", uniq_name = "_QEc"}
+! FIRDialect:         %[[VAL_3:.*]] = fir.alloca i32 {bindc_name = "i", uniq_name = "_QEi"}
+! FIRDialect:         %[[VAL_4:.*]] = fir.alloca i32 {bindc_name = "j", uniq_name = "_QEj"}
+! FIRDialect:         %[[VAL_5:.*]] = fir.alloca i32 {bindc_name = "k", uniq_name = "_QEk"}
+! FIRDialect:         %[[VAL_6:.*]] = fir.alloca i32 {bindc_name = "x", uniq_name = "_QEx"}
+! LLVMIRDialect:           %[[VAL_0:.*]] = llvm.mlir.constant(3 : i32) : i32
+! LLVMIRDialect:           %[[VAL_1:.*]] = llvm.mlir.constant(2 : i32) : i32
+! LLVMIRDialect:           %[[VAL_2:.*]] = llvm.mlir.constant(5 : i32) : i32
+! LLVMIRDialect:           %[[VAL_3:.*]] = llvm.mlir.constant(0 : i32) : i32
+! LLVMIRDialect:           %[[VAL_4:.*]] = llvm.mlir.constant(1 : i32) : i32
+! LLVMIRDialect:           %[[VAL_7:.*]] = llvm.mlir.constant(1 : i64) : i64
+  a=3
+! FIRDialect:         %[[VAL_7:.*]] = constant 3 : i32
+! FIRDialect:         fir.store %[[VAL_7]] to %[[VAL_0]] : !fir.ref<i32>
+  b=2
+! FIRDialect:         %[[VAL_8:.*]] = constant 2 : i32
+! FIRDialect:         fir.store %[[VAL_8]] to %[[VAL_1]] : !fir.ref<i32>
+  c=5
+! FIRDialect:         %[[VAL_9:.*]] = constant 5 : i32
+! FIRDialect:         fir.store %[[VAL_9]] to %[[VAL_2]] : !fir.ref<i32>
+  x=0
+! FIRDialect:         %[[VAL_10:.*]] = constant 0 : i32
+! FIRDialect:         fir.store %[[VAL_10]] to %[[VAL_6]] : !fir.ref<i32>
+! LLVMIRDialect:           %[[VAL_8:.*]] = llvm.alloca %[[VAL_7]] x i32 {bindc_name = "a", in_type = i32, operand_segment_sizes = dense<0> : vector<2xi32>, uniq_name = "_QEa"} : (i64) -> !llvm.ptr<i32>
```

Comment:
Here and below, maybe we can reduce the attributes that are being checked to only the uniq_name.